client/src/components/CurrentWeatherFunctions.ts

```diff
@@ -0,0 +1,32 @@
+type Coordinates = {
+  latitude: number;
+  longitude: number;
+};
+
+export async function getCityCoordinates(city: string | undefined) {
+  const result = await fetch(`https://geocoding-api.open-meteo.com/v1/search?name=${city}`);
```

Comment:
doporucil bych 
```
const result = await fetch(`https://geocoding-api.open-meteo.com/v1/search?name=${encodeURIComponent(city)}`);

```

pokud by to city melo "citlivy" znak v nazvu, ktery ma specialni vyznam pro URL (napr. otaznik, & apod.) tak takto neescapovane by ti to rozbilo vysledne URL. Proto ta funkce `encodeURIComponent` ktera to enkoduje do bezpecneho formatu